src/test/Test.hs

```diff
@@ -78,12 +78,18 @@ typeCheckSMT :: Solver -> GenT (Reader Bool) Property
 typeCheckSMT solver = do
   behv <- genBehv 3
   let smtconf = SMTConfig solver 1 False
-      smt = show . getSMT <$> mkPostconditionQueries (B behv)
-  pure . monadicIO . run $ and . fmap parseOutput <$> mapM (runSMT smtconf) smt
+      smt = mkPostconditionQueries (B behv)
+  pure . monadicIO . run $ runQueries smtconf smt
     where
-      parseOutput (exitCode, stdout, _) = case exitCode of
-        ExitFailure _ -> False
-        ExitSuccess -> not . any (isPrefixOf "(error") . filter (/= "") . lines $ stdout
+      runQueries smtconf queries = do
+        solverInstance <- spawnSolver smtconf
+        res <- mapM (askSMT solverInstance) queries
+        pure $ null (catMaybes res)
+
+      getSMT (Postcondition _ _ smt) = smt
+      getSMT _ = undefined
```

Comment:
Even though the input will always have been run through `mkPostconditionQueries` I would say that using `undefined` is bad practice in production code. (And `mkPostconditionQueries` may not do what the name suggests.) Maybe at least use `error "some nice msg"` instead?